### ISSUE TITLE
Remove Ansible deprecation warnings when the cloud-ec2 provider is chosen

### ISSUE THREAD

**Issue by glennschler** [trailofbits/algo#14282]
**Describe the bug**

There are a few Ansible deprecation warnings when the cloud-ec2 provider is chosen. These warnings were added to the 2.9 version of Ansible when they changed some AWS module names. The minimum version stated in Algo's `requirements.txt` is 2.9.20, so there is no need to leave the old names for backward compatibility. There is indication in the messages that the old names might fail in future versions of Ansible.

- [DEPRECATION WARNING]: `aws_region_facts` is kept for backwards compatibility but usage is discouraged. The module documentation details page may explain more about this rationale.. This feature will be removed in a future release. 
  - Ansible documentation states: This redirect has been deprecated. Please update your tasks to use the new name `community.aws.aws_region_info` instead. It will be removed in a major release after 2021-12-01 of community.aws

- [DEPRECATION WARNING]: `ec2_eip_facts` is kept for backwards compatibility but usage is discouraged. The module documentation details page may explain more about this rationale.
  - Ansible documentation states:  This redirect has been deprecated. Please update your tasks to use the new name `community.aws.ec2_eip_info` instead. It will be removed in a major release after 2021-12-01 of community.aws. 

- [DEPRECATION WARNING]: `ec2_ami_facts` is kept for backwards compatibility but usage is discouraged. The module documentation details page may explain more about this rationale.
  - Ansible documentation states: This redirect has been deprecated. Please update your tasks to use the new name `amazon.aws.ec2_ami_info` instead. It will be removed in a major release after 2021-12-01 of amazon.aws. 

**To Reproduce**

1. Choose "3. Amazon EC2" as the cloud provider
2. Log file will show the deprecation warning shown above

**Expected behavior**

Algo should execute without these deprecation warnings.

**Full log**

```
Algo running on: Ubuntu 20.04.3 LTS
Created from git clone. Last commit: 1c6702d azure regions (#14277)
Python 3.8.10
Runtime variables:
    algo_provider "ec2"
    algo_ondemand_cellular "False"
    algo_ondemand_wifi "False"
    algo_ondemand_wifi_exclude "X251bGw="
    algo_dns_adblocking "False"
    algo_ssh_tunneling "False"
    wireguard_enabled "True"
    dns_encryption "True"

TASK [Display the invocation environment] *************************************************************************************************************************************************************************************************************************
changed: [localhost]

TASK [Install the requirements] ***********************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [Generate the SSH private key] *******************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [Generate the SSH public key] ********************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [Copy the private SSH key to /tmp] ***************************************************************************************************************************************************************************************************************************
ok: [localhost]
[DEPRECATION WARNING]: aws_region_facts is kept for backwards compatibility but usage is discouraged. The module documentation details page may explain more about this rationale. This feature will be removed in a future release. Deprecation warnings can be
disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: ec2_eip_facts is kept for backwards compatibility but usage is discouraged. The module documentation details page may explain more about this rationale.. This feature will be removed in a future release. Deprecation warnings can be
disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: ec2_ami_facts is kept for backwards compatibility but usage is discouraged. The module documentation details page may explain more about this rationale.. This feature will be removed in a future release. Deprecation warnings can be
disabled by setting deprecation_warnings=False in ansible.cfg.
```

**Comment by glennschler** [comment]
Currently testing a fix @glennschler@a906e23079cfc5b107f806f1d6c022e1df88ead3

**Comment by jackivanov** [comment]
It's addressed in #14272
